exchanges/btcmarkets/btcmarkets.go

```diff
@@ -52,316 +75,472 @@ type BTCMarkets struct {
 	}
-	return resp.Orders, nil
+	return resp, b.SendAuthenticatedRequest(http.MethodDelete, btcMarketsOrders, req, &resp)
 }
 
-// GetAccountBalance returns the full account balance
-func (b *BTCMarkets) GetAccountBalance() ([]AccountBalance, error) {
-	var balance []AccountBalance
+// FetchOrder finds order based on the provided id
+func (b *BTCMarkets) FetchOrder(id string) (OrderData, error) {
+	var resp OrderData
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsOrders+"/"+id,
+		nil, &resp)
+}
 
-	err := b.SendAuthenticatedRequest(http.MethodGet, btcMarketsAccountBalance, nil, &balance)
-	if err != nil {
-		return nil, err
-	}
+// RemoveOrder removes a given order
+func (b *BTCMarkets) RemoveOrder(id string) (CancelOrderResp, error) {
+	var resp CancelOrderResp
+	return resp, b.SendAuthenticatedRequest(http.MethodDelete, btcMarketsOrders+"/"+id,
+		nil, &resp)
+}
 
-	// All values are returned in Satoshis, even for fiat currencies.
-	for i := range balance {
-		balance[i].Balance /= common.SatoshisPerBTC
-		balance[i].PendingFunds /= common.SatoshisPerBTC
-	}
-	return balance, nil
+// ListWithdrawals lists the withdrawal history
+func (b *BTCMarkets) ListWithdrawals() ([]TransferData, error) {
+	var resp []TransferData
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsWithdrawals, nil, &resp)
 }
 
-// GetTradingFee returns the account's trading fee for a currency pair
-func (b *BTCMarkets) GetTradingFee(base, quote currency.Code) (TradingFee, error) {
-	var tradingFee TradingFee
-	path := fmt.Sprintf(btcMarketsTradingFee, base, quote)
-	return tradingFee, b.SendAuthenticatedRequest(http.MethodGet, path, nil, &tradingFee)
+// GetWithdrawal gets withdrawawl info for a given id
+func (b *BTCMarkets) GetWithdrawal(id string) (TransferData, error) {
+	var resp TransferData
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsWithdrawals+"/"+id,
+		nil, &resp)
 }
 
-// WithdrawCrypto withdraws cryptocurrency into a designated address
-func (b *BTCMarkets) WithdrawCrypto(amount float64, currency, address string) (string, error) {
-	newAmount := int64(amount * float64(common.SatoshisPerBTC))
+// ListDeposits lists the deposit history
+func (b *BTCMarkets) ListDeposits() ([]TransferData, error) {
+	var resp []TransferData
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsDeposits, nil, &resp)
+}
 
-	req := WithdrawRequestCrypto{
-		Amount:   newAmount,
-		Currency: strings.ToUpper(currency),
-		Address:  address,
-	}
+// GetDeposit gets deposit info for a given ID
+func (b *BTCMarkets) GetDeposit(id string) (TransferData, error) {
+	var resp TransferData
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsDeposits+"/"+id,
+		nil, &resp)
+}
 
-	resp := Response{}
-	err := b.SendAuthenticatedRequest(http.MethodPost,
-		btcMarketsWithdrawCrypto,
-		req,
+// ListTransfers lists the past asset transfers
+func (b *BTCMarkets) ListTransfers() ([]TransferData, error) {
```

Comment:
This seems to be missing optional parameter support:
https://api.btcmarkets.net/doc/v3#tag/Fund-Management-APIs/paths/~1v3~1transfers/get